docs/development/pigs/pig-009.rst

```diff
@@ -0,0 +1,231 @@
+assuming a costant event rate. Finally, the IRF corrections are not applied to sampled
+events.
+
+
+Performance and precision evaluation
+====================================
+To evaluate the precision and performance of the described framework we propose
+to implement a prototype for a simulation / fitting pipeline. Starting from a
+selection of spatial, spectral and temporal models, data is simulated and fitted
+multiple times to derive distributions and pull-distributions of the reconstructed
+model parameters. This pipeline should also monitor the required cpu and memory usage.
+This first prototype can be used to evaluate the optimal bin-size (with the best
+compromise between performance and precision) for the simulations and to verify
+the over-all correctness of the simulated data. Later this prototype can be
+developed further into a full simulation / fitting continuous integration system.
+
+
+Alternatives / Outlook
+======================
+So far Gammapy only supports binned likelihood analysis and technically most use-
+cases for the event sampling could be solved with binned simulations. A binned
+simulation can be basically achieved by a call to `numpy.random.poisson()` based
+on the predicted number of counts map. This is conceptionally simpler as well as
+computationally more efficient then a sampling of event lists. However, to support
+the full data access and data reduction process for simulations, event lists are
+required. In future Gammapy possibly also supports event based analysis methods
+(unbinned likelihood, but also e.g. clustering algorithms), that also require event
+lists. For this reason binned simulations cannot present a full equivalent
+solution to event sampling.
+
+The question of the API to simulate multiple observations from e.g. an `ObservationTable`
+or a list of `GTIs` as it is needed for simulating data for the CTA data challenge
+is not addressed in this PIG. For the scope of this PIG, the fundamental class 
+`MapDatasetEventSampler` to simulate events corresponding to a given observation 
+and/or single GTI is in place.
+
+The proposed Event Sampler will not provide, for each event, the corresponding
+`DETX` and `DETY` position. These will be added in a future development of the 
+simulator.
+
+
+List of proposed pull requests
+==============================
+
+This is a proposal for a list of pull requests implementing the proposed changes, 
+ordered by priority:
+
+ 1. Implement the `sample` method in `gammapy.cube.MapDataset` and add tests (v0.14).
+ 2. Implement the `sample` method in `gammapy.time.models.LightCurveTableModel` and `gammapy.time.models.PhaseCurveTableModel` and add tests (v0.14).
+ 3. Implement the `distribute` method in `gammapy.cube.PSFMap` and add tests (v0.14).
+ 4. Implement the `distribute` method in `gammapy.cube.EdispMap` and add tests (v0.14).
+ 5. Introduce the `MapDatasetEventSampler` into `gammapy.cube.` and add tests (v.1.0).
```

Comment:
Your task 5 is listed as v1.0, and then your task 6 as v0.14.
But I think task 6 depends on task 5, so this doesn't work.

suggest to just remove all of the mentions of versions when you want to have this done. we don't rely on having this in a given version, so it's not important to specify here.

Reply:
Done